server/bin/pbench-index.py

```diff
@@ -87,9 +85,7 @@ def main(options, name):
             7 - Bad metadata.log file encountered
             8 - Error reading a mapping file for Elasticsearch templates
             9 - Error creating one of the Elasticsearch templates
-           10 - Bad hostname in a sosreport
-           11 - Failure unpacking the tar ball
-           12 - generic error, needs to be investigated and can be retried
+            10 - generic error, needs to be investigated and can be retried
```

Comment:
We don't want to change the meaning of the generic error, because we track this value on-disk currently.  Best to leave this as `12`, and just remove references to `10` and `11`.

Reply:
And perhaps instead of encoding this is comments, we turn this into a dictionary, where we use the constants you define below as keys to a dictionary containing the strings they map to for the exception.  Then we can reference those strings algorithmically when building log messages from the exceptions.

Reply:
ok

Reply:
And Pete's comment got me back to thinking about this...

This could also be a decent use for an Enum, or an IntEnum (which would allow exiting with the integer equivalent). (Although Python enum isn't as flexible -- or as easily abused? -- as Java enum!)

For example,

```
class ExitCode(IntEnum):
    OK = 0                            # Successful completion
    OP_ERROR = 1             # Operational error while indexing
    CONFIG_FILE = 2          # Configuration file not specified
    BAD_CFG = 2                 # Bad configuration file
```
In fact, if I was doing this in Java I might even extend the enum constructor to include a default message... not so much because we need it, but because it would move the documentation commentary into actual code. And it'd be easy to add an "ignorable" property, too. It's not clear that's feasible in Python, though.

An alternative that would combine all the meta-data might be something like this:

```
class ErrorCode:
    def __init__(self, name, value, ignorable, message):
        self.name = name
        self.value = value
        self.ignorable = ignorable
        self.message = message

class Errors:
    def __init__(self, *codes):
        self.errors = dict()
        for code in codes:
            self.errors[error.name] = error

    def __getitem__(self, key):
        return self.errors[key]

error_codes = Errors(
    ErrorCode('OK', 0, True, 'Successful completion'),
    ErrorCode('OP_ERROR', 1, False, 'Operational error during indexing'),
    [...]
    )
```

Reply:
seems like a good approach